synapse/federation/federation_base.py

```diff
@@ -129,6 +129,13 @@ async def _check_sigs_and_hash(
                     "event_id": pdu.event_id,
                 }
             )
+
+            # TODO: Is it okay to assume this is called from backfilling?
```

Comment:
It's called from many places. It doesn't necessarily hurt to add extra attempts.

But we might not need it here because it will fail later on `_process_pulled_event`

Reply:
Removed for now. Still curious about the spam checker interaction with backfill and will investigate that soon, https://github.com/matrix-org/synapse/pull/13588#discussion_r951989868